sdk/clientcore/annotation-processor-test/pom.xml

```diff
@@ -78,25 +78,65 @@
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
         <version>3.13.0</version> <!-- {x-version-update;org.apache.maven.plugins:maven-compiler-plugin;external_dependency} -->
-        <configuration>
-          <proc>only</proc>
-          <!-- Supported source version 'RELEASE_8' from annotation processor 'io.clientcore.annotation.processor.AnnotationProcessor' less than -source '17' -->
-          <failOnWarning>false</failOnWarning>
-          <generatedSourcesDirectory>${project.build.directory}/generated-sources/</generatedSourcesDirectory>
-          <annotationProcessorPaths>
-            <annotationProcessorPath>
-              <groupId>io.clientcore</groupId>
-              <artifactId>annotation-processor</artifactId>
-              <version>1.0.0-beta.1</version> <!-- {x-version-update;io.clientcore:annotation-processor;current} -->
-            </annotationProcessorPath>
-          </annotationProcessorPaths>
-          <annotationProcessors>
-            <annotationProcessor>io.clientcore.annotation.processor.AnnotationProcessor</annotationProcessor>
-          </annotationProcessors>
-          <compileSourceRoots>
-            <compileSourceRoot>${project.build.directory}/generated-sources/</compileSourceRoot>
-          </compileSourceRoots>
-        </configuration>
+        <executions>
+          <execution>
+            <id>run-annotation-processing</id>
+            <phase>generate-sources</phase>
+            <goals>
+              <goal>compile</goal>
+            </goals>
+
+            <configuration>
+              <source>1.8</source>
+              <target>1.8</target>
+              <release>8</release>
+              <proc>only</proc>
+              <generatedSourcesDirectory>${project.build.directory}/generated-sources/</generatedSourcesDirectory>
+              <annotationProcessorPaths>
+                <annotationProcessorPath>
+                  <groupId>io.clientcore</groupId>
+                  <artifactId>annotation-processor</artifactId>
+                  <version>1.0.0-beta.1</version> <!-- {x-version-update;io.clientcore:annotation-processor;current} -->
+                </annotationProcessorPath>
+              </annotationProcessorPaths>
+              <annotationProcessors>
+                <annotationProcessor>io.clientcore.annotation.processor.AnnotationProcessor</annotationProcessor>
+              </annotationProcessors>
+
+              <compilerArgs>
+                <arg>-Xlint:-options</arg>
+              </compilerArgs>
+            </configuration>
+          </execution>
+        </executions>
+
+        <dependencies>
```

Comment:
Does it not work as expected if we don't add this plugin dependency?

Reply:
Without this Maven doesn't prevent `annotation-processor-test` from being built while `annotation-processor` is being built which can cause build failures when building in parallel. I don't know why this is the case, but this does make this necessary.

Without this we'd need to make the build way more complex by needing to know whether we're doing a non-From Source or From Source build then determining when annotation-processor-test can run based on that.